tests/Feature/OrganisationsTest.php

```diff
@@ -691,10 +698,209 @@ public function test_organisation_admin_can_delete_logo()
+        $response->assertStatus(Response::HTTP_OK);
+
+        $this->assertDatabaseHas(table(UpdateRequest::class), ['updateable_id' => $organisation->id]);
+        $updateRequest = UpdateRequest::where('updateable_id', $organisation->id)->firstOrFail();
+        $this->assertEquals($address->id, $updateRequest->data['location_id']);
+
+        $updateRequest->apply($user);
+
+        $this->assertDatabaseHas(table(Organisation::class), [
+            'id' => $organisation->id,
+            'location_id' => $address->id,
+        ]);
+    }
+
+    /**
+     * @test
+     */
+    public function organisation_admin_can_update_address()
+    {
+        $faker = Faker::create('en_GB');
+        /**
+         * @var \App\Models\User $user
+         */
+        $user = factory(User::class)->create();
+        $user->makeGlobalAdmin();
+        $organisation = factory(Organisation::class)->states('location')->create();
+        $address1 = $organisation->location;
+        $address2 = factory(Location::class)->create();
+
+        Passport::actingAs($user);
+
+        $payload = [
+            'location_id' => $address2->id,
+        ];
+
+        $response = $this->json('PUT', "/core/v1/organisations/{$organisation->id}", $payload);
+        $response->assertStatus(Response::HTTP_OK);
+
+        $this->assertDatabaseHas(table(UpdateRequest::class), ['updateable_id' => $organisation->id]);
+        $updateRequest = UpdateRequest::where('updateable_id', $organisation->id)->firstOrFail();
+        $this->assertEquals($address2->id, $updateRequest->data['location_id']);
+
+        $updateRequest->apply($user);
+
+        $this->assertDatabaseHas(table(Organisation::class), [
+            'id' => $organisation->id,
+            'location_id' => $address2->id,
+        ]);
+    }
+
+    /**
+     * @test
+     */
+    public function organisation_admin_can_delete_address()
+    {
+        $faker = Faker::create('en_GB');
+        /**
+         * @var \App\Models\User $user
+         */
+        $user = factory(User::class)->create();
+        $user->makeGlobalAdmin();
+        $organisation = factory(Organisation::class)->states('location')->create();
+        $address = $organisation->location;
+
+        Passport::actingAs($user);
+
+        $payload = [
+            'location_id' => null,
+        ];
+
+        $response = $this->json('PUT', "/core/v1/organisations/{$organisation->id}", $payload);
+        $response->assertStatus(Response::HTTP_OK);
+
+        $this->assertDatabaseHas(table(UpdateRequest::class), ['updateable_id' => $organisation->id]);
+        $updateRequest = UpdateRequest::where('updateable_id', $organisation->id)->firstOrFail();
+        $this->assertEmpty($updateRequest->data['location_id']);
+
+        $updateRequest->apply($user);
+
+        $this->assertDatabaseMissing(table(Organisation::class), [
+            'id' => $organisation->id,
+            'location_id' => $address->id,
+        ]);
```

Comment:
Could we do this to make the test a bit more explicit?

```suggestion
        $this->assertDatabaseHas(table(Organisation::class), [
            'id' => $organisation->id,
            'location_id' => null,
        ]);
```

Reply:
Done.